### PR TITLE
Add arch flags for Intel GPU Ponte Vecchio

### PR DESCRIPTION
**Pull request by masterleinad**
Among others, announcements for this GPU to be used in Aurora can be found at https://www.intel.com/content/www/us/en/newsroom/news/intel-technology-roadmaps-milestones.html.

### CODE REVIEW

**Review by dalg24 — COMMENTED** [review]
What about the GNU makefiles?

**Review by dalg24 — COMMENTED** [review]
add to the `[gnu_]generate_makefile.bash` scripts

**Review by crtrott — APPROVED** [review]

**Review by dalg24 — COMMENTED** [review]
No ` -D__STRICT_ANSI__` with the GNU makefiles?

**Comment by masterleinad** [comment]
> No ` -D__STRICT_ANSI__` with the GNU makefiles?

That's for `OpenMPTarget` which doesn't have support for Intel architectures in Makefiles yet. I would prefer to do that in a separate pull request.

**Comment by masterleinad** [comment]
I decided to also push the changes to add OpenMPTarget Intel GPU architectures flags to Makefile.kokkos. I'm happy to discuss the approach separately and remove the last commit here, though.

File: Makefile.kokkos [hunk 1]
```diff
@@ -1242,47 +1247,62 @@ ifeq ($(KOKKOS_INTERNAL_USE_HIP), 1)
...
-    tmp := $(call kokkos_append_header,"$H""define KOKKOS_ARCH_INTEL_GPU")
-    tmp := $(call kokkos_append_header,"$H""define KOKKOS_ARCH_INTEL_DG1")
-    KOKKOS_INTERNAL_SYCL_ARCH_FLAG := -fsycl-targets=spir64_gen-unknown-unknown-sycldevice -Xsycl-target-backend "-device dg1"
-  endif
-  ifeq ($(KOKKOS_INTERNAL_USE_ARCH_INTEL_XEHP), 1)
-    tmp := $(call kokkos_append_header,"$H""define KOKKOS_ARCH_INTEL_GPU")
-    tmp := $(call kokkos_append_header,"$H""define KOKKOS_ARCH_INTEL_XEHP")
-    KOKKOS_INTERNAL_SYCL_ARCH_FLAG := -fsycl-targets=spir64_gen-unknown-unknown-sycldevice -Xsycl-target-backend "-device xehp"
-  endif
+  KOKKOS_INTERNAL_LC_BACKEND := sycl
```
Comment (dalg24) [inline Makefile.kokkos:1252]: What does "LC" stand for?

**Review by dalg24 — APPROVED** [review]

File: Makefile.kokkos (hunk 1 above)
Comment (masterleinad) [reply Makefile.kokkos:1252]: lowercase

**Comment by masterleinad** [comment]
Only `CUDA-9.2-NVCC` is failing in `default_exec.overlap_mdrange_policy` which is clearly unrelated.
`

**Pull request merged**